### ISSUE TITLE
Constraining functions to be monotonic

### ISSUE THREAD

**Issue by hftsoi** [hftsoi/symbolfit#3]
There is a way to constrain all the output functions to be monotonically increasing or decreasing by setting a custom loss_function in PySR, which penalizes based on the gradient, sth like
```
loss_function = """
function eval_loss(tree, dataset::Dataset{T,L}, options)::L where {T,L}
    prediction, gradient, flag = eval_grad_tree_array(tree, dataset.X, options)
    if !flag
        return L(Inf)
    end

    prediction      # [nrows]
    gradient        # [1, nrows]
    dataset.y       # [nrows]
    dataset.weights # [nrows]

    loss = sum(
        gi -> gi > 0 ? abs2(gi) : zero(gi),
        gradient
    )
    loss += sum(
        i -> abs2(prediction[i] - dataset.y[i]) * dataset.weights[i],
        eachindex(prediction)
    )
    return loss
end
"""
```
Can be useful for HEP, where background distributions are often smoothly falling in mass. Should experiment this and add a simple flag choice to SymbolFit().
